File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        ans = 0
        l = 0 
        cnt = 0
        for r in range(len(nums)):
            if nums[r] == 0:
                cnt+=1
            while(cnt>1):
                if(nums[l]==0):
                    cnt-=1
                l+=1
            ans=max(ans,r-l)
        return ans

File: 1493-longest-subarray-of-1s-after-deleting-one-element/1493-longest-subarray-of-1s-after-deleting-one-element.java
class Solution {
    public int longestSubarray(int[] nums) {
        int count =0,l=0,ans=0;
        for(int r =0;r<nums.length;r++){
            if(nums[r]==0) count++;
            while(count>1){
                if(nums[l]==0) count--;
                l++;
            }
            ans = Math.max(ans,r-l);
        }
        return ans;
    }
}